File: frontend/src/microservices/tour-group/objects/TourGroupChatSettingsResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupChatSettingsResponse`，作为响应数据并提供 JSON 编解码。

export type TourGroupChatSettingsResponse = {
  groupId: string
  allowMemberDirectChat: boolean
  updatedAt: string
  updatedByUserId: string
  canUpdate: boolean
}
export const tourGroupChatSettingsResponseFromJson = (json: string): TourGroupChatSettingsResponse =>
  JSON.parse(json) as TourGroupChatSettingsResponse

export const tourGroupChatSettingsResponseToJson = (value: TourGroupChatSettingsResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupChatSettingsResponse.scala
// TourGroupChatPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import com.typesafe.travel.shared.kernel.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

import java.time.Instant

final case class TourGroupChatSettingsResponse(
    groupId: String,
    allowMemberDirectChat: Boolean,
    updatedAt: String,
    updatedByUserId: String,
    canUpdate: Boolean
)
object TourGroupChatSettingsResponse:
  given sourceEncoder: Encoder[TourGroupChatSettingsResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupChatSettingsResponse] = deriveDecoder
